### PR TITLE
[#6007] fix: fix hadoop catalog entity name limitation

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

 - limit '/' in schema name and fileset name for hadoop catalog
 - fix name encode issue when create schema, table, fileset and topic

### Why are the changes needed?

Fix: #6007 

### Does this PR introduce _any_ user-facing change?

yes, add new limitation for fileset name

### How was this patch tested?

ITs added

### CODE REVIEW

File: catalogs/catalog-hadoop/src/test/java/org/apache/gravitino/catalog/hadoop/integration/test/HadoopCatalogIT.java [hunk 1]
```diff
@@ -301,12 +288,34 @@ public void testExternalFileset() throws IOException {
...
-        NoSuchFilesetException.class, () -> catalog.asFilesetCatalog().loadFileset(nameIdentifier));
+    exception =
+        Assertions.assertThrows(
+            IllegalArgumentException.class,
+            () ->
+                catalog
+                    .asFilesetCatalog()
+                    .createFileset(nameIdentifier, null, MANAGED, null, null));
+    Assertions.assertTrue(
+        exception.getMessage().contains("does not support '/' in the name for FILESET"));
```
Comment (jerryshao) [inline catalogs/catalog-hadoop/src/test/java/org/apache/gravitino/catalog/hadoop/integration/test/HadoopCatalogIT.java:318]: Can you please also add a test to verify the fileset rename to a illegal name?

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogCapability.java [hunk 2]
```diff
@@ -31,4 +31,19 @@ public CapabilityResult managedStorage(Scope scope) {
...
   }
+
+  @Override
+  public CapabilityResult specificationOnName(Scope scope, String name) {
+    CapabilityResult capabilityResult = Capability.super.specificationOnName(scope, name);
+    if (!capabilityResult.supported()) {
+      return capabilityResult;
+    }
+
+    if (name.contains("/")) {
```
Comment (tengqm) [inline catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogCapability.java:42]: Thanks for adding this check.
I'm thinking if we could promote things like this to project-wide naming constraints.
In Kubernetes, for example, we always check if a given name is a valid DNS subdomain name, sometimes we even require a DNS label. The rationale behind this is that we don't want unexpected characters in names that appear in URIs. They are simply too dangerous. And ... there is always a bound for name lengths.

File: catalogs/catalog-hadoop/src/test/java/org/apache/gravitino/catalog/hadoop/integration/test/HadoopCatalogIT.java (hunk 1 above)
Comment (mchades) [reply catalogs/catalog-hadoop/src/test/java/org/apache/gravitino/catalog/hadoop/integration/test/HadoopCatalogIT.java:325]: added

File: catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogCapability.java (hunk 2 above)
Comment (mchades) [reply catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogCapability.java:42]: What you described is exactly the responsibility of interface `Capability`:
https://github.com/apache/gravitino/blob/9a5fef924ee3488ef99bb00c75913f885a84dff3/core/src/main/java/org/apache/gravitino/connector/capability/Capability.java#L74-L83

which will be implemented by each catalog. It already has a default implementation with constraints on length and special characters. I think it is considered strict enough.

**Comment by jerryshao** [comment]
Should we also update the Python client to encode the path?

**Comment by mchades** [comment]
> Should we also update the Python client to encode the path?

yeah, added

**Review by jerryshao — APPROVED** [review]

**Pull request merged**